sources/web/datalab/polymer/components/datalab-data/datalab-data.ts

```diff
@@ -18,208 +18,170 @@
+        observer: '_renderResultsList',
         type: Array,
         value: () => [],
       },
-      _tablesFilterValue: {
+      _searchValue: {
         type: String,
-        value: 'no-filter',
-      },
-      _tablesList: {
-        observer: '_renderTablesList',
-        type: Array,
-        value: () => [],
+        value: '',
       },
     };
   }
 
   ready() {
     super.ready();
 
-    this._collectionsList = [];
-    const collectionsElement = this.$.collections as ItemListElement;
-    if (collectionsElement) {
-      collectionsElement.addEventListener('itemDoubleClick',
-                                    this._collectionsDoubleClicked.bind(this));
-      collectionsElement.addEventListener('selected-indices-changed',
-                                    this._collectionsSelectionChanged.bind(this));
+    this._resultsList = [];
+    const resultsElement = this.$.results as ItemListElement;
+    if (resultsElement) {
+      resultsElement.addEventListener('itemDoubleClick',
+                                    this._resultsDoubleClicked.bind(this));
+      resultsElement.addEventListener('selected-indices-changed',
+                                    this._resultsSelectionChanged.bind(this));
     }
 
-    this._tablesList = [];
-    const tablesElement = this.$.tables as ItemListElement;
-    if (tablesElement) {
-      tablesElement.addEventListener('itemDoubleClick',
-                                    this._tablesDoubleClicked.bind(this));
-      tablesElement.addEventListener('selected-indices-changed',
-                                    this._tablesSelectionChanged.bind(this));
-    }
+    (this.$.results as ItemListElement).columns = ['Name', 'Type'];
+    this.$.searchKeys.target = this.$.searchBox;
   }
 
-  _generateFakeCollectionsListForTesting() {
-    const collectionsList = [];
-    const count = Math.floor(Math.random() * 20);
-    for (let i = 0; i < count; i++) {
-      collectionsList.push(this._generateFakeCollectionForTesting());
-    }
-    return collectionsList;
+  /** Sends the user's query to the search API, renders results as they get returned. */
+  _search() {
+    this._resultsList = [];
+    this._sendQuery(this._searchValue, this._handleQueryResults.bind(this));
   }
 
-  _generateFakeCollectionForTesting() {
-    const fakeCollection = {
-      name: 'fakeCollection' + Math.floor(Math.random() * 1000000),
-      status: 'fake',
-    };
-    return fakeCollection;
+  _sendQuery(searchValue: string, resultHandler: (partialResults: Result[]) => void) {
+    this._callBigQuery(searchValue, resultHandler);
   }
 
-  _generateFakeTablesListForTesting() {
-    const tablesList = [];
-    const count = Math.floor(Math.random() * 20);
-    for (let i = 0; i < count; i++) {
-      tablesList.push(this._generateFakeTableForTesting());
-    }
-    return tablesList;
+  _handleQueryResults(partialResults: Result[]) {
+    this._resultsList = this._resultsList.concat(partialResults);
```

Comment:
Yeah, that could happen. I added a TODO to look at that. This is all part of the messy search API that I need to figure out and fix. This code is all pretty much POC at the moment.